api.js

```diff
@@ -43,6 +43,20 @@ class Api extends EventEmitter {
 		const apiOptions = this.options;
 		runtimeOptions = runtimeOptions || {};
 
+		const {extensions: doNotCompileExtensions = []} = apiOptions;
+		const {extensions: babelExtensions = []} = apiOptions.babelConfig || {};
+
+		if (!apiOptions.babelConfig && doNotCompileExtensions.length === 0) {
+			doNotCompileExtensions.push('js');
+		}
+
+		if (apiOptions.babelConfig && babelExtensions.length === 0) {
+			babelExtensions.push('js');
+		}
+
+		// Combine all extensions possible for testing. Removing duplicate extensions.
+		const allExtensions = [...new Set([...doNotCompileExtensions, ...babelExtensions])];
```

Comment:
This just needs to ensure there is no overlap between both arrays. You could compare the length of `allExtensions` to the combined length of `doNotCompileExtensions` and `babelExtensions`: it should be equal.

We should probably include the ambiguous extensions in the error message though. Some applications of [`lodash.difference`](https://www.npmjs.com/package/lodash.difference) should do the trick.

Given that we're now throwing an error, could you move this logic into the constructor? Feel free to save `allExtensions` on the instance.